src/java/main/org/apache/zookeeper/Login.java

```diff
@@ -90,14 +90,14 @@
      * @throws javax.security.auth.login.LoginException
      *               Thrown if authentication fails.
      */
-    public Login(final String loginContextName, CallbackHandler callbackHandler)
+    public Login(final String loginContextName, CallbackHandler callbackHandler, final ZKConfig config)
             throws LoginException {
         this.callbackHandler = callbackHandler;
         login = login(loginContextName);
         this.loginContextName = loginContextName;
         subject = login.getSubject();
         isKrbTicket = !subject.getPrivateCredentials(KerberosTicket.class).isEmpty();
-        AppConfigurationEntry entries[] = Configuration.getConfiguration().getAppConfigurationEntry(loginContextName);
+        AppConfigurationEntry entries[] = javax.security.auth.login.Configuration.getConfiguration().getAppConfigurationEntry(loginContextName);
```

Comment:
yes, by creating different sections in JAAS file and changing the value of the zookeeper.sasl.clientconfig